### DISCUSSION TITLE
Updated to latest firmware, how to roll back

### DISCUSSION THREAD

**Discussion by rkrueger11** [gtxaspec/wz_mini_hacks#556] [Q&A]
I'm on the latest firmware on my Pan v2.. 4.49.1.47

How can I use this to expose RTSP on my camera? It is impossible to downgrade to a firmware version that works with this?

**Comment by virmaior** [comment]
not super familiar with the Pan v2, but the general procedure is to use the SD card method and load the previous version. The most recent versions encrypt the ssid and pw so rolling back requires resetting everything afterwards.

Assuming this is the same, to expose the RTSP vis wz_mini would take the same thing.

**Comment by rkrueger11** [comment]
@virmaior What do you mean by "encrypt the ssid and pw"? and where can I find the previous firmware versions, since they're no longer provided by Wyze?

**Comment by virmaior** [comment]
> @virmaior What do you mean by "encrypt the ssid and pw"? 

in versions before the latest for V2/V3, the ssid and password for wifi are stored in raw text in `/configs` as `.wifissid` and `.wifipasswd` respectively
In the latest versions they are in the same places but are encrypted -- meaning if you roll back a V2/V3 from the latest firmware, it needs to be setup again.

> and where can I find the previous firmware versions, since they're no longer provided by Wyze?

Someone has them posted in a discussion or issue here. I don't remember who/where.

**Reply by akivacp** [reply]
https://github.com/gtxaspec/wz_mini_hacks/discussions/351#discussioncomment-4583811

**Reply by akivacp** [reply]
I was trying to find the Pan v2 firmware and I found out I posted it myself.
